PayPalNative/src/main/java/com/braintreepayments/api/PayPalNativeClient.java

```diff
@@ -13,33 +31,44 @@
 
     private final BraintreeClient braintreeClient;
     private final PayPalInternalClient internalPayPalClient;
+    private final TokenizationClient tokenizationClient;
+
     private PayPalClient payPalClient;
 
     public PayPalNativeClient(BraintreeClient braintreeClient) {
         this.braintreeClient = braintreeClient;
         this.internalPayPalClient = new PayPalInternalClient(braintreeClient);
+        this.tokenizationClient = new TokenizationClient(braintreeClient);
     }
 
-    public void tokenizePayPalAccount(final FragmentActivity activity, final PayPalRequest request, final PayPalNativeFlowStartedCallback callback) {
+    public void tokenizePayPalAccount(final FragmentActivity activity, final PayPalRequest request, final PayPalNativeFlowCallback callback) {
         if (request instanceof PayPalNativeCheckoutRequest) {
             sendNativeCheckoutRequest(activity, (PayPalNativeCheckoutRequest) request, callback);
         } else if (request instanceof PayPalNativeVaultRequest) {
             sendVaultRequest(activity, (PayPalNativeVaultRequest) request, callback);
         }
     }
 
-    private void sendNativeCheckoutRequest(final FragmentActivity activity, final PayPalNativeCheckoutRequest request, final PayPalNativeFlowStartedCallback callback) {
+    private void sendNativeCheckoutRequest(final FragmentActivity activity, final PayPalNativeCheckoutRequest request, final PayPalNativeFlowCallback callback) {
+        //This is copied from PayPal module.
+        braintreeClient.sendAnalyticsEvent("paypal.native.single-payment.selected");
+        if (request.getShouldOfferPayLater()) {
+            braintreeClient.sendAnalyticsEvent("paypal.native.single-payment.paylater.offered");
+        }
+
         braintreeClient.getConfiguration(new ConfigurationCallback() {
             @Override
             public void onResult(Configuration configuration, Exception error) {
-                if (error != null) {
-                    callback.onResult(createPayPalNotEnabledError());
+
+                if (payPalConfigInvalid(configuration) || error != null) {
+                    Exception configInvalidError = createPayPalNotEnabledError();
+                    callback.onResult(null, configInvalidError);
                     return;
                 }
 
                 String payPalClientId = configuration.getPayPalClientId();
                 if (payPalClientId == null) {
-                    callback.onResult(new BraintreeException("PayPal Client ID is null. TODO: write error message explaining why PayPal Native couldn't be started"));
+                    callback.onResult(null, new BraintreeException("PayPal Client ID is null. TODO: write error message explaining why PayPal Native couldn't be started"));
```

Comment:
Just calling out that we should clarify this exception message and remove the TODO